### DISCUSSION TITLE
`highlight clear SignColumn` isn't respected in .vimrc

### DISCUSSION THREAD

**Discussion by ghost** [airblade/vim-gitgutter#820] [Q&A]
I've recently setup with git gutter and am having issues with the SignColumn. The last lines in my `.vimrc` are 
```vim
let g:gitgutter_set_sign_backgrounds = 0 
let g:gitgutter_sign_added = '++'
let g:gitgutter_sign_modified = '~~'
let g:gitgutter_sign_removed = '--'
let g:gitgutter_sign_removed_first_line = '^^'
let g:gitgutter_sign_removed_above_and_below = '^v'
let g:gitgutter_sign_modified_removed = 'vv'

highlight GitGutterAdd    guifg=#00CC00 ctermfg=2
highlight GitGutterChange guifg=#CCCC00 ctermfg=3
highlight GitGutterDelete guibg=#11FF22 ctermfg=1

highlight clear SignColumn
```

But the SignColumn always has its grey background when launching into vim.  If I source my `.vimrc` directly after launch, it gets cleared, but it is always there on boot.   I've tried adding `highlight! link SignColumn LineNr` and `let g:gitgutter_sign_allow_clobber = 1`, but neither seems to work. Has anyone else had this issue?

I'm running Vim 8.2 with GitGutter at commit `18d129`

**Comment by airblade** [comment]
Which colorscheme do you use?

Where in your vimrc do you set it (and possibly `background`) – before or after your `highlight` commands?

**Reply by ghost** [reply]
I don't explicitly set any color scheme.  I stripped down my .vimrc to this
```vim
execute pathogen#infect()
execute pathogen#helptags()

set updatetime=100
let mapleader = "\\"

let g:gitgutter_set_sign_backgrounds = 0
let g:gitgutter_sign_added = '++'
let g:gitgutter_sign_modified = '~~'
let g:gitgutter_sign_removed = '--'
let g:gitgutter_sign_removed_first_line = '^^'
let g:gitgutter_sign_removed_above_and_below = '^v'
let g:gitgutter_sign_modified_removed = 'vv'

highlight GitGutterAdd    guifg=#00CC00 ctermfg=2
highlight GitGutterChange guifg=#CCCC00 ctermfg=3
highlight GitGutterDelete guibg=#11FF22 ctermfg=1

highlight clear SignColumn
```
And am still encountering the same issue

**Comment by airblade** [comment]
If the whole sign column has a grey background, that's the default colorscheme – gitgutter doesn't set the sign column's background.

You could try `syntax enable` at the top of your config.

I used your config on my Vim 8.2.3008 and it behaved as expected, i.e. the sign column's background was the same as the window's background (black).  Which patch version of 8.2 are you on?

You can find out where you sign column is being configured with `:verbose highlight SignColumn`.  What does it say?

**Reply by ghost** [reply]
`syntax enable` didn't manage to get the result I was looking for, but `set background=dark` did.  


For completeness, this is what the verbose output was after 1. directly opening vim and 2. explicitly reloading my `.vimrc` file

Opening a file directly
```
:verbose highlight SignColumn
SignColumn     xxx term=standout ctermfg=14 ctermbg=242 guifg=Cyan guibg=Grey
```

Explicitly running `:source ~/.vimrc`
```
:verbose highlight SignColumn
SignColumn     xxx cleared
        Last set from ~/.dotfiles/vim/vimrc line 19
```

It appears that with the `set background=dark`, it does end up loading it the way I would expect.

**Reply by airblade** [reply]
> It appears that with the `set background=dark`, it does end up loading it the way I would expect.

Good news!
